sync-partials.js

```diff
@@ -5,13 +5,20 @@ import fs from "fs";
 // <!-- INCLUDE path-to-partial.md -->
 // <!-- END -->
 
-const includeRegex = /(\<\!\-\- INCLUDE (.*.md) -->)(.|\n)*(<!-- END -->)/g;
+const includeRegex = /(\<\!\-\- INCLUDE (.*) -->)(.|\n)*?(<!-- END -->)/g;
 const warning =
-  "<!-- WARNING: copied from the included markdown path. Do not edit directly. -->";
+  "<!-- WARNING: copied from the included file path. Do not edit directly. -->";
 
 // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_the_replacement
-function replacer(_, includeTag, markdownFile, __, endTag) {
-  const content = fs.readFileSync(markdownFile, "utf8");
+function replacer(match, includeTag, file, __, endTag) {
+  console.log(`${match}\n\n\n`);
+  const body = fs.readFileSync(file, "utf8");
+  let content;
+  if (file.endsWith(".md")) {
+    content = body;
+  } else if (file.endsWith(".sol")) {
+    content = `\`\`\`solidity\n${body}\n\`\`\``;
```

Comment:
What's the use case here? Which files do you expect to be included?

Reply:
mostly interfaces but also maybe examples like this one which is out of date https://docs.hyperlane.xyz/docs/protocol/sovereign-consensus#interchain-security-modules